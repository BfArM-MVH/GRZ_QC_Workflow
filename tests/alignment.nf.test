nextflow_pipeline {

    name "Test Workflow main.nf"
    script "../main.nf"
    config "../conf/tests/test_alignment.config"

    test("Alignement Test, case senario: input=test-dataset-alignments, 2 samples of WES, tumor+germline, bam input, 1 sample has fastp_json") {

        when {
            params {
                outdir = "${outputDir}"
            }
        }

        then {
            def stable_path = getAllFilesFromDir(params.outdir, relative:true, ignoreFile: 'tests/.nftignore')
            def report = path("${params.outdir}/report.csv")

            assertAll(
                { assert workflow.success },
                { assert snapshot(
                    // Number of successful tasks
                    workflow.trace.succeeded().size(),
                    // All stable path name, with a relative path
                    stable_path,
                    // All files with stable contents
                    report
                ).match() }
            )
        }
    }

}
