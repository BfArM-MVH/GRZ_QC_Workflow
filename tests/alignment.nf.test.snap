{
    "Alignement Test, case senario: input=test-dataset-alignments, 2 samples of WES, tumor+germline, bam input, 1 sample has fastp_json": {
        "content": [
            22,
            [
                "multiqc/multiqc_data/BETA-multiqc.parquet",
                "multiqc/multiqc_data/mosdepth-coverage-per-contig-single.txt",
                "multiqc/multiqc_data/mosdepth-cumcoverage-dist-id.txt",
                "multiqc/multiqc_data/mosdepth_cov_dist.txt",
                "multiqc/multiqc_data/mosdepth_cumcov_dist.txt",
                "multiqc/multiqc_data/mosdepth_perchrom.txt",
                "multiqc/multiqc_data/multiqc.log",
                "multiqc/multiqc_data/multiqc_citations.txt",
                "multiqc/multiqc_data/multiqc_data.json",
                "multiqc/multiqc_data/multiqc_general_stats.txt",
                "multiqc/multiqc_data/multiqc_grz_qc.txt",
                "multiqc/multiqc_data/multiqc_samtools_flagstat.txt",
                "multiqc/multiqc_data/multiqc_samtools_stats.txt",
                "multiqc/multiqc_data/multiqc_software_versions.txt",
                "multiqc/multiqc_data/multiqc_sources.txt",
                "multiqc/multiqc_data/samtools-flagstat-pct-table.txt",
                "multiqc/multiqc_data/samtools-flagstat-table.txt",
                "multiqc/multiqc_data/samtools-stats-dp.txt",
                "multiqc/multiqc_data/samtools_alignment_plot.txt",
                "multiqc/multiqc_plots/pdf/grz_qc.pdf",
                "multiqc/multiqc_plots/pdf/mosdepth-coverage-per-contig-single-cnt.pdf",
                "multiqc/multiqc_plots/pdf/mosdepth-coverage-per-contig-single-pct.pdf",
                "multiqc/multiqc_plots/pdf/mosdepth-cumcoverage-dist-id.pdf",
                "multiqc/multiqc_plots/pdf/samtools-flagstat-pct-table.pdf",
                "multiqc/multiqc_plots/pdf/samtools-flagstat-table.pdf",
                "multiqc/multiqc_plots/pdf/samtools-stats-dp.pdf",
                "multiqc/multiqc_plots/pdf/samtools_alignment_plot-cnt.pdf",
                "multiqc/multiqc_plots/pdf/samtools_alignment_plot-pct.pdf",
                "multiqc/multiqc_plots/png/grz_qc.png",
                "multiqc/multiqc_plots/png/mosdepth-coverage-per-contig-single-cnt.png",
                "multiqc/multiqc_plots/png/mosdepth-coverage-per-contig-single-pct.png",
                "multiqc/multiqc_plots/png/mosdepth-cumcoverage-dist-id.png",
                "multiqc/multiqc_plots/png/samtools-flagstat-pct-table.png",
                "multiqc/multiqc_plots/png/samtools-flagstat-table.png",
                "multiqc/multiqc_plots/png/samtools-stats-dp.png",
                "multiqc/multiqc_plots/png/samtools_alignment_plot-cnt.png",
                "multiqc/multiqc_plots/png/samtools_alignment_plot-pct.png",
                "multiqc/multiqc_plots/svg/grz_qc.svg",
                "multiqc/multiqc_plots/svg/mosdepth-coverage-per-contig-single-cnt.svg",
                "multiqc/multiqc_plots/svg/mosdepth-coverage-per-contig-single-pct.svg",
                "multiqc/multiqc_plots/svg/mosdepth-cumcoverage-dist-id.svg",
                "multiqc/multiqc_plots/svg/samtools-flagstat-pct-table.svg",
                "multiqc/multiqc_plots/svg/samtools-flagstat-table.svg",
                "multiqc/multiqc_plots/svg/samtools-stats-dp.svg",
                "multiqc/multiqc_plots/svg/samtools_alignment_plot-cnt.svg",
                "multiqc/multiqc_plots/svg/samtools_alignment_plot-pct.svg",
                "report.csv",
                "report.xlsx",
                "s1_Blood_DNA_sample/alignments/s1_Blood_DNA_sample.md.bam",
                "s1_Blood_DNA_sample/compare_threshold/s1_Blood_DNA_sample.result.csv",
                "s1_Blood_DNA_sample/mosdepth/s1_Blood_DNA_sample.mosdepth.summary.txt",
                "s1_Blood_DNA_sample/mosdepth/s1_Blood_DNA_sample.per-base.bed.gz",
                "s1_Blood_DNA_sample/mosdepth/s1_Blood_DNA_sample.regions.bed.gz",
                "s1_Blood_DNA_sample_less_samples/alignments/s1_Blood_DNA_sample_less_samples.md.bam",
                "s1_Blood_DNA_sample_less_samples/compare_threshold/s1_Blood_DNA_sample_less_samples.result.csv",
                "s1_Blood_DNA_sample_less_samples/mosdepth/s1_Blood_DNA_sample_less_samples.mosdepth.summary.txt",
                "s1_Blood_DNA_sample_less_samples/mosdepth/s1_Blood_DNA_sample_less_samples.per-base.bed.gz",
                "s1_Blood_DNA_sample_less_samples/mosdepth/s1_Blood_DNA_sample_less_samples.regions.bed.gz"
            ],
            "report.csv:md5,e8433201e3316d515338e4c3b2869a4c"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.3"
        },
        "timestamp": "2025-07-21T15:25:51.467457167"
    }
}